api/v1alpha1/common_types.go

```diff
@@ -591,6 +602,61 @@ type VaultSecretReference struct {
 	// The Vault key of the secret
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	Key string `json:"key"`
+	// The Vault secret store reference
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	SecretStoreRef *VaultSecretStoreReferenceSpec `json:"secretStoreRef,omitempty"`
+	// The Vault refresh interval of the secret (seconds)
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	RefreshInterval *metav1.Duration `json:"refreshInterval,omitempty"`
+}
+
+func (spec *VaultSecretReference) Default() {
+	spec.SecretStoreRef = InitializeVaultSecretStoreReferenceSpec(spec.SecretStoreRef, defaultVaultSecretStoreReference)
+	spec.RefreshInterval = durationOrDefault(spec.RefreshInterval, &defaultVaultRefreshInterval)
+}
+
+// VaultSecretStoreReferenceSpec is a reference to a secret store
+type VaultSecretStoreReferenceSpec struct {
+	// The Vault secret store reference name
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// The Vault secret store reference kind
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Kind *string `json:"kind,omitempty"`
+}
+
+type defaultVaultSecretStoreReferenceSpec struct {
+	Name, Kind *string
+}
+
+// Default sets default values for any value not specifically set in the VaultSecretStoreReferenceSpec struct
+func (spec *VaultSecretStoreReferenceSpec) Default(def defaultVaultSecretStoreReferenceSpec) {
+	spec.Name = stringOrDefault(spec.Name, def.Name)
+	spec.Kind = stringOrDefault(spec.Kind, def.Kind)
+}
+
+// IsDeactivated true if the field is set with the deactivated value (empty struct)
+func (spec *VaultSecretStoreReferenceSpec) IsDeactivated() bool {
+	return reflect.DeepEqual(spec, &VaultSecretStoreReferenceSpec{})
+}
```

Comment:
Does the `IsDeactivated()` semantic makes sense for this field? If I understand it correctly, the `VaultSecretStoreReferenceSpec` can either be left with default vaules or be set explicitely by the user, but I don't see how it could be "deactivated" as it is something that the ExternalSecret resource always requires.

Reply:
Yes that's true! so It needs just to be completely deleted right??

Reply:
You also need to modify the function `InitializeVaultSecretStoreReferenceSpec()`. See the [suggestion](https://github.com/3scale-ops/saas-operator/pull/150/files#r785878167) I added to that function.

Reply:
```suggestion
```

Reply:
Thanks! I have applied both suggestions